src/main/java/org/opensearch/ad/model/AnomalyDetector.java

```diff
@@ -587,7 +586,7 @@ public TimeConfiguration getWindowDelay() {
     }
 
     public Integer getShingleSize() {
-        return isMultientityDetector() ? DEFAULT_MULTI_ENTITY_SHINGLE : shingleSize;
+        return shingleSize;
```

Comment:
General question for supporting shingle size: do we need to re-train model if user changed shingle size? The old checkpoint is using different shingle size, can we restore model from checkpoint and feed feature data with different shingle size into it?

Reply:
For existing detector, their shingle size is recorded in the config: https://github.com/opensearch-project/anomaly-detection/blob/main/src/main/java/org/opensearch/ad/model/AnomalyDetector.java#L319

Example:

```
{
    "_id": "rjCrensBDqnenptucFOY",
    "_version": 1,
    "_primary_term": 1,
    "_seq_no": 0,
    "anomaly_detector": {
        "name": "Second Test Detector",
        "description": "ok rate",
        "time_field": "@timestamp",
        "indices": [
            "server-metrics"
        ],
        "filter_query": {
            "match_all": {
                "boost": 1.0
            }
        },
        "detection_interval": {
            "period": {
                "interval": 1,
                "unit": "Minutes"
            }
        },
        "window_delay": {
            "period": {
                "interval": 10,
                "unit": "Seconds"
            }
        },
        "shingle_size": 1,
        "schema_version": 2,
        "feature_attributes": [
            {
                "feature_id": "deny_max",
                "feature_name": "deny max",
                "feature_enabled": true,
                "aggregation_query": {
                    "deny_max": {
                        "max": {
                            "field": "deny"
                        }
                    }
                }
            }
        ],
        "ui_metadata": {
            "aabb": {
                "ab": "bb"
            }
        },
        "last_update_time": 1629850661013,
        "category_field": [
            "host"
        ]
    },
    "anomaly_detector_job": null,
    "realtime_detection_task": null,
    "historical_analysis_task": null
}
```

When reading it out, it is still one: https://github.com/opensearch-project/anomaly-detection/blob/main/src/main/java/org/opensearch/ad/model/AnomalyDetector.java#L457

So the detector is still using shingle size 1.  No conversion or bwc for them.  

Is it correct?

Reply:
Oh, my question is about RCF model checkpoint. Before AD 1.1, the RCF model checkpoint is for model with shingle size 1. Not sure if we will delete all model checkpoint when user stop detector and update. If not, the model checkpoint may not be restored and used for new shingle size.